src/KurrentDB.Client/Core/Common/Diagnostics/ActivitySourceExtensions.cs

```diff
@@ -54,10 +56,69 @@ public static void TraceSubscriptionEvent(
 				userCredentials?.Username ?? settings.DefaultCredentials?.Username
 			);
 
-		StartActivity(source, TracingConstants.Operations.Subscribe, ActivityKind.Consumer, tags, parentContext)
+		StartActivity(source, Operations.Subscribe, ActivityKind.Consumer, tags, parentContext)
 			?.Dispose();
 	}
 
+	public static async IAsyncEnumerable<(Activity? Activity, AppendStreamRequest Request)> InstrumentAppendOperations(
+		this ActivitySource source,
+		IAsyncEnumerable<AppendStreamRequest> requests,
+		Func<AppendStreamRequest, ActivityTagsCollection> createTags
+	) {
+		var currentActivity = Activity.Current;
+		var startTime = DateTime.UtcNow;
+
+		try {
+			Activity.Current = null;
```

Comment:
This basically prevents parent context propagation so the trace gets broken.

Reply:
We don't need to start an activity per message. 

> A **span** represents a unit of work or operation.
> [Source](https://opentelemetry.io/docs/concepts/signals/traces/#spans)

We need _one_ activity to be started at the beginning of `MultiStreamAppendAsync` in a usual way and then dispose it before returning the result. The activity error should be set of the result is not success. The operation should be different from `streams.append` to indicate it's not a regular append. Maybe use 'streams.multi-append` or something. We won't set tags for stream name, event id and event type unless we decided to concatenate strings or add event data as span events.